src/spikeinterface/comparison/groundtruthstudy.py

```diff
@@ -1,327 +1,405 @@
-
         perf_by_unit = pd.concat(perf_by_unit)
-        perf_by_unit = perf_by_unit.set_index(["rec_name", "sorter_name", "gt_unit_id"])
-
+        perf_by_unit = perf_by_unit.set_index(self.levels)
         return perf_by_unit
 
-    def aggregate_count_units(self, well_detected_score=None, redundant_score=None, overmerged_score=None):
-        assert self.comparisons is not None, "run_comparisons first"
+    def get_count_units(
+            self, case_keys=None, well_detected_score=None, redundant_score=None, overmerged_score=None
+        ):
 
         import pandas as pd
 
-        index = pd.MultiIndex.from_tuples(self.computed_names, names=["rec_name", "sorter_name"])
+        if case_keys is None:
+            case_keys = list(self.cases.keys())
+
+        if isinstance(case_keys[0], str):
+            index = pd.Index(case_keys, name=self.levels)
+        else:
+            index = pd.MultiIndex.from_tuples(case_keys, names=self.levels)
+
+
+        columns = ["num_gt", "num_sorter", "num_well_detected", "num_redundant", "num_overmerged"]
+        comp = self.comparisons[case_keys[0]]
+        if comp.exhaustive_gt:
+            columns.extend(["num_false_positive", "num_bad"])
+        count_units = pd.DataFrame(index=index, columns=columns, dtype=int)
 
-        count_units = pd.DataFrame(
-            index=index,
-            columns=["num_gt", "num_sorter", "num_well_detected", "num_redundant", "num_overmerged"],
-            dtype=int,
-        )
 
-        if self.exhaustive_gt:
-            count_units["num_false_positive"] = pd.Series(dtype=int)
-            count_units["num_bad"] = pd.Series(dtype=int)
+        for key in case_keys:
+            comp = self.comparisons.get(key, None)
+            assert comp is not None, "You need to do study.run_comparisons() first"
 
-        for rec_name, sorter_name, sorting in iter_computed_sorting(self.study_folder):
-            gt_sorting = self.get_ground_truth(rec_name)
-            comp = self.comparisons[(rec_name, sorter_name)]
+            gt_sorting = comp.sorting1
+            sorting = comp.sorting2
 
-            count_units.loc[(rec_name, sorter_name), "num_gt"] = len(gt_sorting.get_unit_ids())
-            count_units.loc[(rec_name, sorter_name), "num_sorter"] = len(sorting.get_unit_ids())
-            count_units.loc[(rec_name, sorter_name), "num_well_detected"] = comp.count_well_detected_units(
+            count_units.loc[key, "num_gt"] = len(gt_sorting.get_unit_ids())
+            count_units.loc[key, "num_sorter"] = len(sorting.get_unit_ids())
+            count_units.loc[key, "num_well_detected"] = comp.count_well_detected_units(
                 well_detected_score
             )
-            if self.exhaustive_gt:
-                count_units.loc[(rec_name, sorter_name), "num_overmerged"] = comp.count_overmerged_units(
+            if comp.exhaustive_gt:
+                count_units.loc[key, "num_overmerged"] = comp.count_overmerged_units(
```

Comment:
Above (line 375) only `num_false_positive` and `num_bad` are added for exhaustive GT. Should `num_overmerged` also be added only in that case? For `redundant`, I think it can be computed also wthout exhaustive, no?